blaze-server/src/main/scala/org/http4s/server/blaze/Http2NodeStage.scala

```diff
@@ -224,18 +225,20 @@ private class Http2NodeStage[F[_]](
       val hs = HHeaders(headers.result())
       val req = Request(method, path, HttpVersion.`HTTP/2.0`, hs, body, attributes())
       executionContext.execute(new Runnable {
-        def run(): Unit = {
-          val action = Sync[F]
-            .defer(raceTimeout(req))
+        def run(): Unit =
+          F.defer(raceTimeout(req))
             .recoverWith(serviceErrorHandler(req))
-            .flatMap(renderResponse)
-
-          F.runAsync(action) {
-            case Right(()) => IO.unit
-            case Left(t) =>
-              IO(logger.error(t)(s"Error running request: $req")).attempt *> IO(closePipeline(None))
-          }
-        }.unsafeRunSync()
+            .continual {
```

Comment:
_[Just trying to figure things out]_: What actually could cause interruption if we are run execution via `Runnable` here? Is it some sort of JVM signals?

Reply:
That's a good question.  In `Http1ServerStage`, we expose a cancel token outside the Runnable, but I don't see how this one might be canceled.  (Which is probably something this stage ought to do!)

Reply:
I don't think this hurts anything, and leaves us in a better place if anyone ever adds proper cancellation support.  (Though I can't imagine it happening on CE2...)